yjit/src/backend/ir.rs

```diff
@@ -1083,15 +1083,16 @@ impl Assembler
         let gc_offsets = self.compile_with_regs(cb, alloc_regs);
 
         #[cfg(feature = "disasm")]
-        if let Some(target) = get_option!(dump_disasm) {
-            if target == DumpDisasm::All || !cb.outlined {
+        match get_option!(dump_disasm) {
+            DumpDisasm::All | DumpDisasm::Inlined if cb.inlined() => {
```

Comment:
I think you intended `DumpDisasm::All | (DumpDisasm::Inlined if cb.inlined())`, but it seems like it's actually evaluated like `(DumpDisasm::All | DumpDisasm::Inlined) if cb.inlined()`, so this doesn't print outlined cb.

Reply:
🤔 https://github.com/ruby/ruby/pull/6309/commits/d4e2c19ffa2f337a53c5f3a62f7afac9a7baa72e